h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoost.java

```diff
@@ -600,40 +363,67 @@ protected final void scoreAndBuildTrees(XGBoostModel model, DMatrix trainMat, DM
         model._output._training_time_ms = ArrayUtils.copyAndFillOf(model._output._training_time_ms, model._output._ntrees+1, System.currentTimeMillis());
         if (stop_requested() && !timeout()) throw new Job.JobCancelledException();
         if (timeout()) { //stop after scoring
-          if (!scored) doScoring(model, model.model_info()._booster, trainMat, validMat, true, tmpModelDir);
+          if (!scored) {
+            doScoring(model, model.model_info().getBooster(), true, tmpModelDir);
+          }
           _job.update(_parms._ntrees-model._output._ntrees); //finish
           break;
         }
       }
-      doScoring(model, model.model_info()._booster, trainMat, validMat, true, tmpModelDir);
+      doScoring(model, model.model_info().getBooster(), true, tmpModelDir);
+    }
+
+    // Don't start the tracker for 1 node clouds -> the GPU plugin fails in such a case
+    private void startRabitTracker(RabitTracker rt) {
```

Comment:
No worries! It's all quite hacky on their (xgboost/rabit) side - Rabit has a `tracker.py` script bundled inside the jar, RabitTracker is just a wrapper that extracts that file to local disk and simply runs it `Runtime.getRuntime().exec("python " + tracker_py + " --log-level=DEBUG --num-workers=" + this.numWorkers)` plus does some process sysout scraping etc. So it's just whatever `python` is available from the child process.

They don't have a native Java tracker impl unfortunately (only that Python wrapper and a Scala impl).